EIP-5069: EIP editor handbook in EIP form (#5069)
ethereum/EIPs: EIPS/eip-5069.md

---
eip: 5069
title: EIP Editor Handbook

> description: Formalizes the process for becoming an EIP editor

Reviewer: ```suggestion
description: Handy reference for EIP editors and those who want to become one
```

If this is a handbook for editors, the description should reflect that.

author: Pooja Ranjan (@poojaranjan), Pandapip1 (@Pandapip1)
discussions-to: https://ethereum-magicians.org/t/pr-5069-eip-editor-handbook/9137
status: Living
type: Informational
created: 2022-05-02
requires: 1
---

## Abstract
An Ethereum Improvement Proposal (EIP) is a design document providing information to the Ethereum community, or describing a new feature for Ethereum or its processes or environment. The EIP standardization process is a mechanism for proposing new features, for collecting community technical input on an issue, and for documenting the design decisions that have gone into Ethereum. Because improvement proposals are key components of Ethereum blockchain, it is important that they are well reviewed before reaching `Final` status. EIPs are stored in text files in a versioned repository which is monitored by the EIP editors.

This EIP describes the recommended process for becoming an EIP editor.

## Specification
The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

### Application and Onboarding Process

> Any contributor of Ethereum ecosystem, having a good understanding of EIP standardization and network upgrade process, intermediate level experience on the core and/or application side of the Ethereum blockchain, and willingness to contribute to the process management MAY apply to become an EIP editor. Potential EIP editors SHOULD have the following skills:

Reviewer: Just cleaning up the wording a little bit:

```suggestion
Anyone having a good understanding of the EIP standardization and network upgrade process, intermediate level experience on the core and/or application side of the Ethereum blockchain, and willingness to contribute to the process management MAY apply to become an EIP editor. Potential EIP editors SHOULD have the following skills:
```

- Good communication skills
- Ability to handle contentious discourse
- 1-5 spare hours per week

The best available resource to understand the EIP process is [EIP-1](./eip-1.md). Anyone desirous of becoming an EIP editor MUST understand this document. Afterwards, participating in the EIP process by commenting on and suggesting improvements to PRs and issues will familliarize the procedure, and is RECOMMENDED. The contributions of newer editors SHALL be monitored by other EIP editors.

> A pull request with a request to be listed as EIP editor MAY be made by anyone who feels comfortable with the EIP process. If the existing EIP editors agree, the author SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list SHALL be made.

Reviewer: ```suggestion
Anyone meeting the above requirements MAY make a pull request adding themselves as an EIP editor. If the existing EIP editors approve, the author SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list SHALL be made.
```

Changed the wording a bit, but I'm still not sure what the second pull request is for?

The following are external links to example pull requests:

> - Add Sam Wilson as an editor: https://github.com/ethereum/EIPs/pull/4859

Reviewer: I think you could just include a patch snippet here showing the diff:

```patch
diff --git a/EIPS/eip-1.md b/EIPS/eip-1.md
index 79558a3..079b196 100644
--- a/EIPS/eip-1.md
+++ b/EIPS/eip-1.md
@@ -207,6 +211,7 @@ The current EIP editors are
 - Matt Garnett (@lightclient)
 - Micah Zoltu (@MicahZoltu)
 - Greg Colvin (@gcolvin)
+- Sam Wilson (@SamWilsn)
 
 Emeritus EIP editors are 
 
diff --git a/.github/workflows/auto-merge-bot.yml b/.github/workflows/auto-merge-bot.yml
index 5730343..3d162ea 100644
--- a/.github/workflows/auto-merge-bot.yml
+++ b/.github/workflows/auto-merge-bot.yml
@@ -17,12 +17,12 @@ jobs:
         id: auto-merge-bot
         with:
           GITHUB-TOKEN: ${{ secrets.TOKEN }} 
-          CORE_EDITORS: "@MicahZoltu,@lightclient,@axic,@gcolvin"
-          ERC_EDITORS: "@lightclient,@axic"
-          NETWORKING_EDITORS: "@MicahZoltu,@lightclient,@axic"
-          INTERFACE_EDITORS: "@lightclient,@axic"
-          META_EDITORS: "@lightclient,@axic,@gcolvin"
-          INFORMATIONAL_EDITORS: "@lightclient,@axic,@gcolvin"
+          CORE_EDITORS: "@MicahZoltu,@lightclient,@axic,@gcolvin,@SamWilsn"
+          ERC_EDITORS: "@lightclient,@axic,@SamWilsn"
+          NETWORKING_EDITORS: "@MicahZoltu,@lightclient,@axic,@SamWilsn"
+          INTERFACE_EDITORS: "@lightclient,@axic,@SamWilsn"
+          META_EDITORS: "@lightclient,@axic,@gcolvin,@SamWilsn"
+          INFORMATIONAL_EDITORS: "@lightclient,@axic,@gcolvin,@SamWilsn"
           MAINTAINERS: "@alita-moore,@mryalamanchi"
   enable-auto-merge:
     if: github.repository == 'ethereum/eips'
```

This EIP describes the recommended process for becoming an EIP editor.

## Specification
The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

### Application and Onboarding Process
Anyone having a good understanding of the EIP standardization and network upgrade process, intermediate level experience on the core and/or application side of the Ethereum blockchain, and willingness to contribute to the process management MAY apply to become an EIP editor. Potential EIP editors SHOULD have the following skills:
- Good communication skills
- Ability to handle contentious discourse
- 1-5 spare hours per week

The best available resource to understand the EIP process is [EIP-1](./eip-1.md). Anyone desirous of becoming an EIP editor MUST understand this document. Afterwards, participating in the EIP process by commenting on and suggesting improvements to PRs and issues will familliarize the procedure, and is RECOMMENDED. The contributions of newer editors SHALL be monitored by other EIP editors.

> Anyone meeting the above requirements MAY make a pull request adding themselves as an EIP editor. If the existing EIP editors approve, the author SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list SHALL be made:

Reviewer: Ah, I see. I'd recommend just making a single pull request (which is what the diff shows)

This Meta-EIP describes the process for becoming an EIP editor.

## Motivation
The process for becoming an EIP editor is previously undocumented in the EIP repository itself.

## Specification
The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

### All PRs
EIP editors MUST do the following for all new PRs submitted to the EIP repository:
1. Review proposals: All updated proposals MUST pass a readiness check, unless this is unfeasable:
    * The title describing the content MUST be accurate.
    * The idea put forth MUST make sense technically.

>     * The content MUST be not contain any language errors (spelling, grammar, and sentence structure).

Reviewer 2: ```suggestion
    * The content MUST not contain any language errors (spelling, grammar, and sentence structure).
```
Fix grammar

    * The proposal MUST use GitHub-flavored Markdown for formatting.
    * The proposal MUST NOT have any blank sections, TODOs, or placeholders.
    * The proposal MUST provide all required sections, and MUST NOT contain any sections not listed in [EIP-1](./eip-1.md).
    * Any code provided MUST use best-practices for that programming language.
    * The proposal MUST follow all latest [EIP-1](./eip-1.md) guidelines.
1. Provide feedback: A review MUST be submitted, containing specific instructions to reject or improve the changes.
1. Approve proposal: When the PR is ready to be merged, a review accepting the changes will result in the PR being merged.

### New Proposals
EIP editors MUST do the following for all new proposals submitted to the EIP repository:
1. Ensure relevance: New proposals MUST be related to the Ethereum ecosystem.

> 1. Assign EIP numbers: New proposals SHALL have an EIP number assigned, which SHOULD be the PR number. In some cases, to discourage sqautting, an unused EIP number SHOULD be assigned instead.

Reviewer 2: ```suggestion
1. Assign EIP numbers: New proposals SHALL have an EIP number assigned, which SHOULD be the PR number. In some cases, to discourage squatting, an unused EIP number SHOULD be assigned instead.
```
Fix typo

- Good communication skills
- Ability to handle contentious discourse
- 1-5 spare hours per week

The best available resource to understand the EIP process is [EIP-1](./eip-1.md). EIP editor apprentices MUST understand this document. Afterwards, participating in the EIP process by commenting on and suggesting improvements to PRs and issues will familliarize apprentices with the procedure, and is RECOMMENDED. Attending EIPIP meetings <!-- is it okay to provide external links for this? --> is also RECOMMENDED, as it rapidly accelerates consensus. The contributions of apprentice editors SHALL be monitored by other EIP editors.

<!-- I'm going to hold off on including this if/until ECH and ETH merge: https://github.com/ethereum-cat-herders/EIPIP/issues/132 -->
<!-- Following the monthly EIPs Insight page to keep track of upcoming Ethereum Improvement Proposals. Among other resources, ECH produces PEEPanEIP video series for a quick overview of these proposals. -->

> When the apprentice feels comfortable with the process, a pull request with a request to be listed as EIP editor MAY to be made. <!-- Add Sam's PR as an external link as an example? --> If the existing EIP editors agree, the apprentice SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list in SHALL be made.

Reviewer 2: ```suggestion
When the apprentice feels comfortable with the process, a pull request with a request to be listed as EIP editor MAY be made. <!-- Add Sam's PR as an external link as an example? --> If the existing EIP editors agree, the apprentice SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list SHALL be made.
```
Fix grammar

---
eip: 5069
title: EIP Editor "Apprentice" Handbook
description: Formalizes the process for becoming an EIP editor
author: Pooja Ranjan (@poojaranjan), Pandapip1 (@Pandapip1), et al.
discussions-to: https://ethereum-magicians.org/t/pr-5069-eip-editor-apprentice-handbook/9137
status: Living
type: Informational
created: 2022-05-02
requires: 1
---

## Abstract

> An Ethereum Improvement Proposal (EIP) is a design document providing information to the Ethereum community, or describing a new feature for Ethereum or its processes or environment. The EIP standardization process is a mechanism for proposing new features, for collecting community technical input on an issue, and for documenting the design decisions that have gone into Ethereum. Because improvement proposals are key components of Ethereum blockchain, it is important that they are well reviewed before reaching `final` status. EIPs are stored in text files in a versioned repository which is monitored by the EIP editors.

Reviewer 3: ```suggestion
An Ethereum Improvement Proposal (EIP) is a design document providing information to the Ethereum community, or describing a new feature for Ethereum or its processes or environment. The EIP standardization process is a mechanism for proposing new features, for collecting community technical input on an issue, and for documenting the design decisions that have gone into Ethereum. Because improvement proposals are key components of Ethereum blockchain, it is important that they are well reviewed before reaching `Final` status. EIPs are stored in text files in a versioned repository which is monitored by the EIP editors.
```

> This EIP describes the process for becoming an EIP editor.

Reviewer 3: ```suggestion
This EIP describes the recommended process for becoming an EIP editor.
```

Reviewer 3: I would suggest specifying "recommended process" to hope for adoption of the _EIP editor onboarding process_, if pursued as an "Informational" EIP. 

Because as per EIP-1, "Informational EIPs do not necessarily represent Ethereum community consensus or a recommendation, so **users and implementers are free to ignore Informational EIPs or follow their advice**."

EIP editors MUST do the following for all new proposals submitted to the EIP repository:
1. Ensure relevance: New proposals MUST be related to the Ethereum ecosystem.
1. Assign EIP numbers: New proposals SHALL have an EIP number assigned, which SHOULD be the PR number. In some cases, to discourage squatting, an unused EIP number SHOULD be assigned instead.
1. Perform the [All PRs](#all-prs) checklist

### Other Changes
Requirements for other PRs (status change, edits to a `final` EIP) are provided in [EIP-1](./eip-1.md).

### Exceptional Cases / Miscellaneous Notes
- When updating the status from `stagnant` to `draft`, it is up to the original author to bring it back to the status as they find appropriate. It MAY contain the same content as when its state was changed to `stagnant`, but MUST pass the readiness check. Note that the readiness check MAY have changed since the EIP was last updated - the EIP MUST be updated to use the latest standards!
- When a proposal has been in the `stagnant` state for more than 6 months, then `stagnant` MAY be considered its terminal status. However, the original author MAY still update the EIP, changing the status from `stagnant` to `draft`.

> - If a proposal needs to be moved to `final` because the `last call` deadline was exceeded, but all authors are absent, an EIP champion MAY be added to the author list to deal with resulting comments. There are no requirements to become an EIP champion. Alternatively, the `last call` deadline may be extended.

Reviewer 3: ```suggestion
- If a proposal needs to be moved to `Final` because the `Last Call` deadline was exceeded, but all authors are absent, an EIP champion MAY be added to the author list to deal with resulting comments. There are no requirements to become an EIP champion. Alternatively, the `last call` deadline may be extended.
```

- Past EIPIP Meetings: https://www.youtube.com/playlist?list=PL4cwHXAawZxpLrRIkDlBjDUUrGgF91pQw
- Past EIPIP Meeting Transcripts: https://github.com/ethereum-cat-herders/EIPIP

When the apprentice feels comfortable with the process, a pull request with a request to be listed as EIP editor MAY be made. If the existing EIP editors agree, the apprentice SHALL become a full EIP editor. This SHALL notify the editor of relevant new proposals submitted in the EIPs repository, and they SHALL review and merge pull requests. Once the first PR is merged, another pull request to update the editor list SHALL be made.

The following are external links to example Pull Requests. These links are not durable.
- Add Sam Wilson as an editor: https://github.com/ethereum/EIPs/pull/4859

### Funding
Ecosystem Support Program provided a grant to ECH to work with present editors and individual applicants to provide a stipend for the purpose of increasing community engagement with the EIP process. Historically, though, EIP editing had been done by volunteers.

### Special Merging Rules for this EIP
This EIP MUST have the same rules regarding changes as [EIP-1](./eip-1.md).

## Rationale

> - "6 months" was chosen as the cutoff for denoting `stagnant` EIPs terminally-`stagnant` arbitrarily.

Reviewer 3: ```suggestion
- "6 months" was chosen as the cutoff for denoting `Stagnant` EIPs terminally-`Stagnant` arbitrarily.
```
